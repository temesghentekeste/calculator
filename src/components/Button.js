import React from 'react';
import PropTypes from 'prop-types';

function Button({ name, clickHandler }) {
  return (
    <div>
      <button type="button" onClick={() => clickHandler(name)}>
        {name}
      </button>
    </div>
  );
}

Button.propTypes = {
  name: PropTypes.string.isRequired,
  clickHandler: PropTypes.func.isRequired,
};

export default Button;
