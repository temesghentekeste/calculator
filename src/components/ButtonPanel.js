/* eslint-disable react/jsx-props-no-spreading */
import React from 'react';
import Button from './Button';

function ButtonPanel(props) {
  return (
    <div>
      <div>
        <Button name="AC" {...props} />
        <Button name="+/-" {...props} />
        <Button name="%" {...props} />
        <Button name="รท" {...props} />
      </div>
      <div>
        <Button name="7" {...props} />
        <Button name="8" {...props} />
        <Button name="9" {...props} />
        <Button name="X" {...props} />
      </div>
      <div>
        <Button name="4" {...props} />
        <Button name="5" />
        <Button name="6" {...props} />
        <Button name="-" {...props} />
      </div>
      <div>
        <Button name="1" {...props} />
        <Button name="2" {...props} />
        <Button name="3" {...props} />
        <Button name="+" {...props} />
      </div>

      <div>
        <Button name="0" {...props} />
        <Button name="." {...props} />
        <Button name="=" {...props} />
      </div>
    </div>
  );
}

export default ButtonPanel;
