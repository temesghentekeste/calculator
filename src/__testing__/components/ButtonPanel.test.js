import React from 'react';
import { render, screen } from '@testing-library/react'
import renderer from 'react-test-renderer';
import '@testing-library/jest-dom/extend-expect';
import ButtonPanel from '../../components/ButtonPanel';
import Button from '../../components/Button';

const handleClick = () => 'Clicked';

it('renders the ButtonPanel component', () => {
  render(<ButtonPanel clickHandler={handleClick}>
    <Button name="AC" clickHandler={handleClick} isLast={true}/>
  </ButtonPanel>);

  expect(screen.queryByText('AC')).toBeTruthy();
});

it('renders the ButtonPanel component with child Button element', () => {
  const component = renderer
    .create(<ButtonPanel clickHandler={handleClick}></ButtonPanel>)
    .toJSON();

  expect(component).toMatchSnapshot();
});
