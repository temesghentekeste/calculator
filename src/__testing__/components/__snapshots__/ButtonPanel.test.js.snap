// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the ButtonPanel component 1`] = `
<div>
  <div
    className="calculatorButtons false"
  >
    <div
      className="calculatorButtonContainer false"
    >
      <button
        className="calculatorButtonContainer__button"
        data-testid="AC"
        onClick={[Function]}
        type="button"
      >
        AC
      </button>
    </div>
    <div
      className="calculatorButtonContainer false"
    >
      <button
        className="calculatorButtonContainer__button"
        data-testid="+/-"
        onClick={[Function]}
        type="button"
      >
        +/-
      </button>
    </div>
    <div
      className="calculatorButtonContainer false"
    >
      <button
        className="calculatorButtonContainer__button"
        data-testid="%"
        onClick={[Function]}
        type="button"
      >
        %
      </button>
    </div>
    <div
      className="calculatorButtonContainer calculatorButtonContainer_last"
    >
      <button
        className="calculatorButtonContainer__button"
        data-testid="รท"
        onClick={[Function]}
        type="button"
      >
        รท
      </button>
    </div>
  </div>
  <div
    className="calculatorButtons false"
  >
    <div
      className="calculatorButtonContainer false"
    >
      <button
        className="calculatorButtonContainer__button"
        data-testid="7"
        onClick={[Function]}
        type="button"
      >
        7
      </button>
    </div>
    <div
      className="calculatorButtonContainer false"
    >
      <button
        className="calculatorButtonContainer__button"
        data-testid="8"
        onClick={[Function]}
        type="button"
      >
        8
      </button>
    </div>
    <div
      className="calculatorButtonContainer false"
    >
      <button
        className="calculatorButtonContainer__button"
        data-testid="9"
        onClick={[Function]}
        type="button"
      >
        9
      </button>
    </div>
    <div
      className="calculatorButtonContainer calculatorButtonContainer_last"
    >
      <button
        className="calculatorButtonContainer__button"
        data-testid="X"
        onClick={[Function]}
        type="button"
      >
        X
      </button>
    </div>
  </div>
  <div
    className="calculatorButtons false"
  >
    <div
      className="calculatorButtonContainer false"
    >
      <button
        className="calculatorButtonContainer__button"
        data-testid="4"
        onClick={[Function]}
        type="button"
      >
        4
      </button>
    </div>
    <div
      className="calculatorButtonContainer false"
    >
      <button
        className="calculatorButtonContainer__button"
        data-testid="5"
        onClick={[Function]}
        type="button"
      >
        5
      </button>
    </div>
    <div
      className="calculatorButtonContainer false"
    >
      <button
        className="calculatorButtonContainer__button"
        data-testid="6"
        onClick={[Function]}
        type="button"
      >
        6
      </button>
    </div>
    <div
      className="calculatorButtonContainer calculatorButtonContainer_last"
    >
      <button
        className="calculatorButtonContainer__button"
        data-testid="-"
        onClick={[Function]}
        type="button"
      >
        -
      </button>
    </div>
  </div>
  <div
    className="calculatorButtons false"
  >
    <div
      className="calculatorButtonContainer false"
    >
      <button
        className="calculatorButtonContainer__button"
        data-testid="1"
        onClick={[Function]}
        type="button"
      >
        1
      </button>
    </div>
    <div
      className="calculatorButtonContainer false"
    >
      <button
        className="calculatorButtonContainer__button"
        data-testid="2"
        onClick={[Function]}
        type="button"
      >
        2
      </button>
    </div>
    <div
      className="calculatorButtonContainer false"
    >
      <button
        className="calculatorButtonContainer__button"
        data-testid="3"
        onClick={[Function]}
        type="button"
      >
        3
      </button>
    </div>
    <div
      className="calculatorButtonContainer calculatorButtonContainer_last"
    >
      <button
        className="calculatorButtonContainer__button"
        data-testid="+"
        onClick={[Function]}
        type="button"
      >
        +
      </button>
    </div>
  </div>
  <div
    className="calculatorButtons calculatorButtons__last"
  >
    <div
      className="calculatorButtonContainer false"
    >
      <button
        className="calculatorButtonContainer__button"
        data-testid="0"
        onClick={[Function]}
        type="button"
      >
        0
      </button>
    </div>
    <div
      className="calculatorButtonContainer false"
    >
      <button
        className="calculatorButtonContainer__button"
        data-testid="."
        onClick={[Function]}
        type="button"
      >
        .
      </button>
    </div>
    <div
      className="calculatorButtonContainer calculatorButtonContainer_last"
    >
      <button
        className="calculatorButtonContainer__button"
        data-testid="="
        onClick={[Function]}
        type="button"
      >
        =
      </button>
    </div>
  </div>
</div>
`;
